### ISSUE TITLE
[ENH] Use logging

### ISSUE THREAD

**Issue by surchs** [neurobagel/bagel-cli#162]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### New feature

Right now we tell the user good news with a print statement, e.g:

https://github.com/neurobagel/bagel-cli/blob/0ed0e4d3fe193ee6d509aed015e85e1cf57de6e4/bagel/cli.py#L70-L74

which means they can't turn that off if they'd prefer (e.g. when they run this thing a couple hundred times in a row).
We should instead use a logger (https://docs.python.org/3/library/logging.html).
This has other nice benefits such as:
- I can choose what level of "news" I'd like to see
- I can have it log to a file 
- and probably others I forget

In short: let's do this.

### Steps to implement
- [x] Add a logger
- [x] Change print statements to INFO
- [x] Change warnings to WARNING
- [x] update tests
- [x] (maybe) remove multiline logs (bad practice)?
- [x] use rich for logs formatting
